src/main/java/org/opensearch/flowframework/transport/GetWorkflowStepTransportAction.java

```diff
@@ -47,11 +53,23 @@ public GetWorkflowStepTransportAction(
     }
 
     @Override
-    protected void doExecute(Task task, ActionRequest request, ActionListener<GetWorkflowStepResponse> listener) {
+    protected void doExecute(Task task, WorkflowRequest request, ActionListener<GetWorkflowStepResponse> listener) {
         try {
-            WorkflowValidator workflowValidator = this.workflowStepFactory.getWorkflowValidator();
+            List<String> steps = request.getParams().size() > 0
+                ? Arrays.asList(Strings.splitStringByCommaToArray(request.getParams().get(WORKFLOW_STEP)))
+                : Collections.emptyList();
+            WorkflowValidator workflowValidator;
+            if (steps.isEmpty()) {
+                workflowValidator = this.workflowStepFactory.getWorkflowValidator();
+            } else {
+                workflowValidator = this.workflowStepFactory.getWorkflowValidatorByStep(steps);
+            }
             listener.onResponse(new GetWorkflowStepResponse(workflowValidator));
         } catch (Exception e) {
+            if (e instanceof FlowFrameworkException) {
+                logger.error("Invalid step name");
```

Comment:
can you also log the exception message here so we record which names were invalid on the request

Reply:
Since we were removing it anyway, that's why I didn't log it: https://github.com/opensearch-project/flow-framework/pull/524

Reply:
We will be reversing changes in #524 to include helpful exception messages and stack traces as well when helpful for debugging purposes

Reply:
Added the exception message.